File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/CommoditySettlementPeriodsPriceSchedule.cpp
// CommoditySettlementPeriodsPriceSchedule.cpp 
#include "CommoditySettlementPeriodsPriceSchedule.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CommoditySettlementPeriodsPriceSchedule::CommoditySettlementPeriodsPriceSchedule(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //settlementPeriodsPriceStepNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* settlementPeriodsPriceStepNode = xmlNode->FirstChildElement("settlementPeriodsPriceStep");

   if(settlementPeriodsPriceStepNode){settlementPeriodsPriceStepIsNull_ = false;}
   else{settlementPeriodsPriceStepIsNull_ = true;}

   if(settlementPeriodsPriceStepNode)
   {
      for(settlementPeriodsPriceStepNode; settlementPeriodsPriceStepNode; settlementPeriodsPriceStepNode = settlementPeriodsPriceStepNode->NextSiblingElement("settlementPeriodsPriceStep")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementPeriodsPriceStepNode , address : " << settlementPeriodsPriceStepNode << std::endl;
          #endif
          if(settlementPeriodsPriceStepNode->Attribute("href") || settlementPeriodsPriceStepNode->Attribute("id"))
          {
              if(settlementPeriodsPriceStepNode->Attribute("id"))
              {
                  settlementPeriodsPriceStepIDRef_ = settlementPeriodsPriceStepNode->Attribute("id");
                  settlementPeriodsPriceStep_.push_back(boost::shared_ptr<FixedPrice>(new FixedPrice(settlementPeriodsPriceStepNode)));
                  settlementPeriodsPriceStep_.back()->setID(settlementPeriodsPriceStepIDRef_);
                  IDManager::instance().setID(settlementPeriodsPriceStepIDRef_, settlementPeriodsPriceStep_.back());
              }
              else if(settlementPeriodsPriceStepNode->Attribute("href")) { settlementPeriodsPriceStepIDRef_ = settlementPeriodsPriceStepNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { settlementPeriodsPriceStep_.push_back(boost::shared_ptr<FixedPrice>(new FixedPrice(settlementPeriodsPriceStepNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementPeriodsPriceStepNode , address : " << settlementPeriodsPriceStepNode << std::endl;
       #endif
   }

   //settlementPeriodsReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* settlementPeriodsReferenceNode = xmlNode->FirstChildElement("settlementPeriodsReference");

   if(settlementPeriodsReferenceNode){settlementPeriodsReferenceIsNull_ = false;}
   else{settlementPeriodsReferenceIsNull_ = true;}

   if(settlementPeriodsReferenceNode)
   {
      for(settlementPeriodsReferenceNode; settlementPeriodsReferenceNode; settlementPeriodsReferenceNode = settlementPeriodsReferenceNode->NextSiblingElement("settlementPeriodsReference")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementPeriodsReferenceNode , address : " << settlementPeriodsReferenceNode << std::endl;
          #endif
          if(settlementPeriodsReferenceNode->Attribute("href") || settlementPeriodsReferenceNode->Attribute("id"))
          {
              if(settlementPeriodsReferenceNode->Attribute("id"))
              {
                  settlementPeriodsReferenceIDRef_ = settlementPeriodsReferenceNode->Attribute("id");
                  settlementPeriodsReference_.push_back(boost::shared_ptr<SettlementPeriodsReference>(new SettlementPeriodsReference(settlementPeriodsReferenceNode)));
                  settlementPeriodsReference_.back()->setID(settlementPeriodsReferenceIDRef_);
                  IDManager::instance().setID(settlementPeriodsReferenceIDRef_, settlementPeriodsReference_.back());
              }
              else if(settlementPeriodsReferenceNode->Attribute("href")) { settlementPeriodsReferenceIDRef_ = settlementPeriodsReferenceNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { settlementPeriodsReference_.push_back(boost::shared_ptr<SettlementPeriodsReference>(new SettlementPeriodsReference(settlementPeriodsReferenceNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementPeriodsReferenceNode , address : " << settlementPeriodsReferenceNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
std::vector<boost::shared_ptr<FixedPrice>> CommoditySettlementPeriodsPriceSchedule::getSettlementPeriodsPriceStep()
{
   if(!this->settlementPeriodsPriceStepIsNull_){
        if(settlementPeriodsPriceStepIDRef_ != ""){
             return this->settlementPeriodsPriceStep_;
        }else{
             return this->settlementPeriodsPriceStep_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<FixedPrice>>();
   }
}
std::vector<boost::shared_ptr<SettlementPeriodsReference>> CommoditySettlementPeriodsPriceSchedule::getSettlementPeriodsReference()
{
   if(!this->settlementPeriodsReferenceIsNull_){
        if(settlementPeriodsReferenceIDRef_ != ""){
             return this->settlementPeriodsReference_;
        }else{
             return this->settlementPeriodsReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<SettlementPeriodsReference>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/CommoditySettlementPeriodsPriceSchedule.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CommoditySettlementPeriodsPriceSchedule : ISerialized
    {
        public CommoditySettlementPeriodsPriceSchedule(XmlNode xmlNode)
        {
            XmlNodeList settlementPeriodsPriceStepNodeList = xmlNode.SelectNodes("settlementPeriodsPriceStep");
            
            if (settlementPeriodsPriceStepNodeList != null)
            {
                this.settlementPeriodsPriceStep_ = new List<FixedPrice>();
                foreach (XmlNode item in settlementPeriodsPriceStepNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            settlementPeriodsPriceStepIDRef_ = item.Attributes["id"].Value;
                            settlementPeriodsPriceStep_.Add(new FixedPrice(item));
                            IDManager.SetID(settlementPeriodsPriceStepIDRef_, settlementPeriodsPriceStep_[settlementPeriodsPriceStep_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            settlementPeriodsPriceStepIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        settlementPeriodsPriceStep_.Add(new FixedPrice(item));
                        }
                    }
                    else
                    {
                        settlementPeriodsPriceStep_.Add(new FixedPrice(item));
                    }
                }
            }
            
        
            XmlNodeList settlementPeriodsReferenceNodeList = xmlNode.SelectNodes("settlementPeriodsReference");
            
            if (settlementPeriodsReferenceNodeList != null)
            {
                this.settlementPeriodsReference_ = new List<SettlementPeriodsReference>();
                foreach (XmlNode item in settlementPeriodsReferenceNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            settlementPeriodsReferenceIDRef_ = item.Attributes["id"].Value;
                            settlementPeriodsReference_.Add(new SettlementPeriodsReference(item));
                            IDManager.SetID(settlementPeriodsReferenceIDRef_, settlementPeriodsReference_[settlementPeriodsReference_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            settlementPeriodsReferenceIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        settlementPeriodsReference_.Add(new SettlementPeriodsReference(item));
                        }
                    }
                    else
                    {
                        settlementPeriodsReference_.Add(new SettlementPeriodsReference(item));
                    }
                }
            }
            
        
        }
        
    
        #region SettlementPeriodsPriceStep_
        private List<FixedPrice> settlementPeriodsPriceStep_;
        public List<FixedPrice> SettlementPeriodsPriceStep_
        {
            get
            {
                if (this.settlementPeriodsPriceStep_ != null)
                {
                    return this.settlementPeriodsPriceStep_; 
                }
                else if (this.settlementPeriodsPriceStepIDRef_ != null)
                {
                    return this.settlementPeriodsPriceStep_; 
                }
                else
                {
                      return this.settlementPeriodsPriceStep_; 
                }
            }
            set
            {
                if (this.settlementPeriodsPriceStep_ != value)
                {
                    this.settlementPeriodsPriceStep_ = value;
                }
            }
        }
        #endregion
        
        public string settlementPeriodsPriceStepIDRef_ { get; set; }
        #region SettlementPeriodsReference_
        private List<SettlementPeriodsReference> settlementPeriodsReference_;
        public List<SettlementPeriodsReference> SettlementPeriodsReference_
        {
            get
            {
                if (this.settlementPeriodsReference_ != null)
                {
                    return this.settlementPeriodsReference_; 
                }
                else if (this.settlementPeriodsReferenceIDRef_ != null)
                {
                    return this.settlementPeriodsReference_; 
                }
                else
                {
                      return this.settlementPeriodsReference_; 
                }
            }
            set
            {
                if (this.settlementPeriodsReference_ != value)
                {
                    this.settlementPeriodsReference_ = value;
                }
            }
        }
        #endregion
        
        public string settlementPeriodsReferenceIDRef_ { get; set; }
        
    
        
    
    }
    
}

